File: data/multilang_comparison/program_go/src/func25.go
package main
import "math/rand"
var _ = rand.Intn
func Func25(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var loop22 int = 0;
   var loopLimit22 int = (50)/3 + 1;
   for ; loop22 < loopLimit22; loop22++ {
      if (PATH0 & 1) != 0 {
         var array69 *Array
         if pCounter > 0 {
             array69 = vars.Data[pCounter-1]
             pCounter--
             array69.RefC++
         } else {
             array69 = &Array{
                 Size: 828,
                 RefC: 1,
                 Id: 69,
                 Data: make([]uint32, 828),
             }
         }
         var loop23 int = 0;
         var loopLimit23 int = (50)/4 + 1;
         for ; loop23 < loopLimit23; loop23++ {
            for i := 0; i < array69.Size; i++ {
                array69.Data[i]--
            }
            for i := 0; i < array69.Size; i++ {
                if array69.Data[i] == 46 {
                    return array69
                }
            }
         }
         var params0 ArrayParam
         params0.Size = 1
         params0.Data = make([]*Array, params0.Size)
         params0.Data[0] = array69
         var array70 *Array = Func42(&params0, getPath(), loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array70.RefC--
         if array70.RefC == 0 {
             array70.Data = nil
             array70 = nil
         }
         array69.RefC--
         if array69.RefC == 0 {
             array69.Data = nil
             array69 = nil
         }
      
      } else {
         var params0 ArrayParam
         params0.Size = 0
         params0.Data = make([]*Array, params0.Size)
         var array71 *Array = Func34(&params0, loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array71.RefC--
         if array71.RefC == 0 {
             array71.Data = nil
             array71 = nil
         }
      }
      var array72 *Array
      if pCounter > 0 {
          array72 = vars.Data[pCounter-1]
          pCounter--
          array72.RefC++
      } else {
          array72 = &Array{
              Size: 810,
              RefC: 1,
              Id: 72,
              Data: make([]uint32, 810),
          }
      }
      var params0 ArrayParam
      params0.Size = 1
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array72
      var array73 *Array = Func35(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      if (PATH0 & 2) != 0 {
         var array74 *Array
         if pCounter > 0 {
             array74 = vars.Data[pCounter-1]
             pCounter--
             array74.RefC++
         } else {
             array74 = &Array{
                 Size: 661,
                 RefC: 1,
                 Id: 74,
                 Data: make([]uint32, 661),
             }
         }
         var loop24 int = 0;
         var loopLimit24 int = (50)/4 + 1;
         for ; loop24 < loopLimit24; loop24++ {
            for i := 0; i < array72.Size; i++ {
                array72.Data[i]--
            }
            for i := 0; i < array73.Size; i++ {
                if array73.Data[i] == 5 {
                    return array73
                }
            }
         }
         array74.RefC--
         if array74.RefC == 0 {
             array74.Data = nil
             array74 = nil
         }
      
      } else {
         var params1 ArrayParam
         params1.Size = 2
         params1.Data = make([]*Array, params1.Size)
         params1.Data[0] = array72
         params1.Data[1] = array73
         var array75 *Array = Func40(&params1, loopsFactor);
         params1.Data = nil
         params1.Size = 0
         array75.RefC--
         if array75.RefC == 0 {
             array75.Data = nil
             array75 = nil
         }
      }
      for i := 0; i < array72.Size; i++ {
          if array72.Data[i] == 26 {
              return array72
          }
      }
      array73.RefC--
      if array73.RefC == 0 {
          array73.Data = nil
          array73 = nil
      }
      array72.RefC--
      if array72.RefC == 0 {
          array72.Data = nil
          array72 = nil
      }
   }
   var array76 *Array
   if pCounter > 0 {
       array76 = vars.Data[pCounter-1]
       pCounter--
       array76.RefC++
   } else {
       array76 = &Array{
           Size: 522,
           RefC: 1,
           Id: 76,
           Data: make([]uint32, 522),
       }
   }
   return array76;
}

File: data/multilang_comparison/program_cpp_array/src/func25.cpp
#include "program_cpp_array.hpp" 
Array* func25(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   unsigned int loop22 = 0;
   unsigned int loopLimit22 = (50)/3 + 1;
   for(; loop22 < loopLimit22; loop22++) {
      if(PATH0 & 1) {
         Array* array69;
         if (pCounter > 0) {
            array69 = vars->data[--pCounter];
            array69->refC++;
            DEBUG_COPY(array69->id);
         } else {
            array69 = new Array();
            array69->size = 828;
            array69->refC = 1;
            array69->id = 69;
            array69->data = new unsigned int[array69->size]();
            DEBUG_NEW(array69->id);
         }
         unsigned int loop23 = 0;
         unsigned int loopLimit23 = (50)/4 + 1;
         for(; loop23 < loopLimit23; loop23++) {
            for (int i = 0; i < array69->size; i++) {
                array69->data[i]--;
            }
            for (int i = 0; i < array69->size; i++) {
                if (array69->data[i] == 46) {
                    return array69;
                }
            }
         }
         Array_param params0;
         params0.size = 1;
         Array* data_params0[1];
         params0.data = data_params0;
         params0.data[0] = array69;
         Array* array70 = func42(&params0, get_path(), loopsFactor);
         DEBUG_RETURN(array70->id);
         if(params0.size > 0) {
         	 params0.data.clear();
         }
         array70->refC--;
         if(array70->refC == 0) {
            delete[] array70->data;
            array70->data = nullptr;
            DEBUG_FREE(array70->id);
            delete array70;
         }
         array69->refC--;
         if(array69->refC == 0) {
            delete[] array69->data;
            array69->data = nullptr;
            DEBUG_FREE(array69->id);
            delete array69;
         }
      }
      else {
         Array_param params0;
         params0.size = 0;
         Array* data_params0[0];
         params0.data = data_params0;
         Array* array71 = func34(&params0, loopsFactor);
         DEBUG_RETURN(array71->id);
         if(params0.size > 0) {
         	 params0.data.clear();
         }
         array71->refC--;
         if(array71->refC == 0) {
            delete[] array71->data;
            array71->data = nullptr;
            DEBUG_FREE(array71->id);
            delete array71;
         }
      }
      Array* array72;
      if (pCounter > 0) {
         array72 = vars->data[--pCounter];
         array72->refC++;
         DEBUG_COPY(array72->id);
      } else {
         array72 = new Array();
         array72->size = 810;
         array72->refC = 1;
         array72->id = 72;
         array72->data = new unsigned int[array72->size]();
         DEBUG_NEW(array72->id);
      }
      Array_param params0;
      params0.size = 1;
      Array* data_params0[1];
      params0.data = data_params0;
      params0.data[0] = array72;
      Array* array73 = func35(&params0, loopsFactor);
      DEBUG_RETURN(array73->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      if(PATH0 & 2) {
         Array* array74;
         if (pCounter > 0) {
            array74 = vars->data[--pCounter];
            array74->refC++;
            DEBUG_COPY(array74->id);
         } else {
            array74 = new Array();
            array74->size = 661;
            array74->refC = 1;
            array74->id = 74;
            array74->data = new unsigned int[array74->size]();
            DEBUG_NEW(array74->id);
         }
         unsigned int loop24 = 0;
         unsigned int loopLimit24 = (50)/4 + 1;
         for(; loop24 < loopLimit24; loop24++) {
            for (int i = 0; i < array72->size; i++) {
                array72->data[i]--;
            }
            for (int i = 0; i < array73->size; i++) {
                if (array73->data[i] == 5) {
                    return array73;
                }
            }
         }
         array74->refC--;
         if(array74->refC == 0) {
            delete[] array74->data;
            array74->data = nullptr;
            DEBUG_FREE(array74->id);
            delete array74;
         }
      }
      else {
         Array_param params1;
         params1.size = 2;
         Array* data_params1[2];
         params1.data = data_params1;
         params1.data[0] = array72;
         params1.data[1] = array73;
         Array* array75 = func40(&params1, loopsFactor);
         DEBUG_RETURN(array75->id);
         if(params1.size > 0) {
         	 params1.data.clear();
         }
         array75->refC--;
         if(array75->refC == 0) {
            delete[] array75->data;
            array75->data = nullptr;
            DEBUG_FREE(array75->id);
            delete array75;
         }
      }
      for (int i = 0; i < array72->size; i++) {
          if (array72->data[i] == 26) {
              return array72;
          }
      }
      array73->refC--;
      if(array73->refC == 0) {
         delete[] array73->data;
         array73->data = nullptr;
         DEBUG_FREE(array73->id);
         delete array73;
      }
      array72->refC--;
      if(array72->refC == 0) {
         delete[] array72->data;
         array72->data = nullptr;
         DEBUG_FREE(array72->id);
         delete array72;
      }
   }
   Array* array76;
   if (pCounter > 0) {
      array76 = vars->data[--pCounter];
      array76->refC++;
      DEBUG_COPY(array76->id);
   } else {
      array76 = new Array();
      array76->size = 522;
      array76->refC = 1;
      array76->id = 76;
      array76->data = new unsigned int[array76->size]();
      DEBUG_NEW(array76->id);
   }
   return array76;
}

